smriprep/workflows/surfaces.py

```diff
@@ -555,3 +567,14 @@ def _check_cw256(in_files):
     if np.any(fov > 256):
         return ['-noskullstrip', '-cw256']
     return '-noskullstrip'
+
+
+def _surfs_meta(surface_file, euler_number):
+    """Creates metadata dictionary for pial, white matter surfaces"""
+    from pathlib import Path
+
+    surface = Path(surface_file).stem
+    if 'smoothwm' in surface or 'pial' in surface:
```

Comment:
Sorry for the late realization. Now I understand part of the problem - again, calculating the Euler number on the smoothwm surface will not be really effective. And we don't generate conversions of the wm without smoothing.

Why don't we attach the Euler number into the `desc-preproc_T1w.json` file, which is always present?

```
{
  "EulerNumbers": {"WM": 1.7, "Pial": 1.9}
}
```

Reply:
ha! - I just had said this ... https://github.com/nipreps/smriprep/pull/215/files#r446595593

Reply:
> calculating the Euler number on the smoothwm surface will not be really effective

Why not? Smoothwm does not change the topology. All surfaces will have an Euler number of 2.

Reply:
I'm confused from your example, the EulerNumber can never be a decimal. 

And, at least from my understanding, the euler number is mainly used a QC check of a surface file - shouldn't this then be tied to the surface file rather than the volumetric data?

Reply:
> I'm confused from your example, the EulerNumber can never be a decimal.

Yup, that's right. I have updated the example with valid values (and also the fact that we have two hemispheres)

> Smoothwm does not change the topology

I believe smoothwm not only includes the smoothing, it also goes through a step to resolve defects. I don't think Rosen et al. used the Euler number from the smoothwm, but rather the wm.

> the euler number is mainly used a QC check of a surface file - shouldn't this then be tied to the surface file rather than the volumetric data?

Correct, but the recent interest in it comes for the fact that it is a good proxy for the image quality of the original T1w image. When I requested this feature, this was my intent. 

That said, there's a little trick in your question: the suffix `_T1w` does not signal whether it is surface or volumetric data (that is hinted/stated by the extension), it signals the measurement type (i.e., physical property). Because these surfaces are reconstructed from the T1w, it would be fair to say the surfaces and corresponding metadata are derived from the T1w (and hence, inherit the suffix).

I think we don't have anything explicit in the BIDS-Derivatives spec when a particular derivative has "two or more parents" (e.g., surface reconstruction with T1w, T2w and FLAIR). I believe using the T1w metadata would be the easier place to find the info, and still compliant with the spec.